snap/snapcraft.yaml

```diff
@@ -25,6 +25,11 @@ layout:
   /usr/lib/$SNAPCRAFT_ARCH_TRIPLET/qemu:
     bind: $SNAP/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/qemu
 
+plugs:
+  all-home:
+    interface: home
+    read: all
```

Comment:
Hmm why would we need this, instead of the plain `home`? (ref. https://snapcraft.io/docs/home-interface)

Reply:
I *think* this is here to de-duplicate some `plugs` definitions for both `multipass` and `multipass-gui`.  I think...

Reply:
Well, sure, but we wouldn't need to deduplicate if we didn't need the `read: all` bit. That's really what I'm referring to.

From https://snapcraft.io/docs/home-interface:
> - `read` (plug):
    optional, when set to ‘all’ allow reading non-hidden files in home directories of all users as traditional file permissions allow.

Why would either `multipass` or `multipass-gui` need to read from `/home/*/` rather than just `$HOME`?

Reply:
Oh, I see what you mean.  But it might be because the mount runs through `multipassd` so it's the `root` user and needs access the user's actual home directory?  Again, I'm not really sure, so I'll have to experiment with it if you think this is really unnecessary.

Reply:
Right after I wrote that, I think I'm wrong :grin: I'll experiment.

Reply:
The only thing the client does is resolve the path, so I don't think that's true. And even if the resolution required access to the mount source, that doesn't cover all the cases for mounts outside of user homes… Some other commands like `transfer` (obviously) and `launch` (for `--cloud-init`) actually access the files client-side, but `read: all` doesn't really help there, does it :)

Reply:
Actually, this is needed to allow `file:\\` based launches.  Without this, we get the issue reported in https://github.com/canonical/multipass/pull/1074#issuecomment-617162865.